examples/pipelines/controlnet-interior-design/components/retrieve_images/src/main.py

```diff
@@ -0,0 +1,89 @@
+"""
+This component retrieves image URLs from LAION-5B based on a set of seed prompts.
+"""
+import logging
+from typing import List
+
+import dask.dataframe as dd
+import dask.array as da
+
+from clip_client import ClipClient, Modality
+
+from fondant.component import FondantTransformComponent
+from fondant.logger import configure_logging
+
+configure_logging()
+logger = logging.getLogger(__name__)
+
+
+def query_clip_client(text: str, client: ClipClient) -> List[str]:
+    """
+    Given a text query and a ClipClient instance, this function retrieves
+    image URLs related to the query.
+
+    Args:
+        text: the text query
+        client (ClipClient): an instance of ClipClient used to query the images
+
+    Returns:
+        results: a list of strings, each representing a URL of an image related to the query
+    """
+    results = client.query(text=text)
+    results = [i["url"] for i in results]
+
+    return results
+
+
+class RetrieveImagesComponent(FondantTransformComponent):
```

Comment:
Perhaps LAIONRetrievalComponent (or something to that effect) might be a better name here, as the component is quite specific to that dataset. Let me know what you think.